homeassistant/components/calendar/__init__.py

```diff
@@ -527,18 +535,28 @@ def event(self) -> CalendarEvent | None:
     @property
     def state_attributes(self) -> dict[str, Any] | None:
         """Return the entity state attributes."""
+        color_hex_val = (
+            f"#{color_rgb_to_hex(*self._attr_color)}" if self._attr_color else None
+        )
         if (event := self.event) is None:
-            return None
+            if color_hex_val is None:
+                return None
+            return {"color": color_hex_val}
```

Comment:
Please add a test for a case where there is no upcoming event but the calendar has a color.

You may just want to add another MockCalendarEntity to the list of entities to exercise this case. You can also pass a color in the constructor instead of changing the attributes in the middle of the test.